src/backend/distributed/executor/multi_utility.c

```diff
@@ -835,6 +859,98 @@ ErrorIfDistributedRenameStmt(RenameStmt *renameStatement)
 
 
 /*
+ * CreateLocalTable gets DDL commands from the remote node for the given
+ * relation. Then, it creates the local relation as temporary and on commit drop.
+ */
+static void
+CreateLocalTable(RangeVar *relation, char *nodeName, int32 nodePort)
+{
+	List *ddlCommandList = NIL;
+	ListCell *ddlCommandCell = NULL;
+	StringInfo tableNameStringInfo = makeStringInfo();
+	char *schemaName = relation->schemaname;
+	char *relationName = relation->relname;
+
+	if (schemaName != NULL)
+	{
+		ereport(ERROR, (errmsg("schema usage is not supported in worker copy")));
+	}
+
+	/* fetch the ddl commands needed to create the table */
+	appendStringInfoString(tableNameStringInfo, relationName);
+
+	/*
+	 * The warning message created in TableDDLCommandList() is descriptive
+	 * enough; therefore, we just throw an error which says that we could not
+	 * run the copy operation.
+	 */
+	ddlCommandList = TableDDLCommandList(nodeName, nodePort, tableNameStringInfo);
+	if (ddlCommandList == NIL)
+	{
+		ereport(ERROR, (errmsg("could not run copy from the worker node")));
+	}
+
+	/* apply DDL commands against the local database */
+	foreach(ddlCommandCell, ddlCommandList)
+	{
+		StringInfo ddlCommand = (StringInfo) lfirst(ddlCommandCell);
+		Node *ddlCommandNode = ParseTreeNode(ddlCommand->data);
+
+		if (IsA(ddlCommandNode, CreateStmt) ||
+			IsA(ddlCommandNode, CreateForeignTableStmt))
+		{
+			CreateStmt *createStatement = NULL;
+
+			if (IsA(ddlCommandNode, CreateStmt))
+			{
+				createStatement = (CreateStmt *) ddlCommandNode;
+			}
+			else
+			{
+				CreateForeignTableStmt *createForeignTableStmt =
+					(CreateForeignTableStmt *) ddlCommandNode;
+				createStatement = (&createForeignTableStmt->base);
+			}
+
+			/* create the local relation as temporary and on commit drop */
+			createStatement->relation->relpersistence = RELPERSISTENCE_TEMP;
+			createStatement->oncommit = ONCOMMIT_DROP;
+		}
+		else if (IsA(ddlCommandNode, AlterTableStmt))
+		{
+			AlterTableStmt *alterTableStatement = (AlterTableStmt *) ddlCommandNode;
+
+			/* make null the default public schema in the alter table statement */
+			alterTableStatement->relation->schemaname = NULL;
```

Comment:
Do we actually need to process these statements? It's only used for setting the primary key at the moment and I don't think we care about the primary key here. To be on the safe side, we could even add an else { continue; } block.